### PR TITLE
Update UI contributing docs

### PR DESCRIPTION
**Pull request by bbovenzi**
Add docs on how to run the new `airflow/ui` project. And add temporary rules during Airflow 3.0 development, specifying when to contribute to the new `airflow/ui` and `airflow/ui_api` projects vs the legacy `airflow/www` UI and endpoints.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: contributing-docs/14_node_environment_setup.rst [hunk 1]
```diff
@@ -18,6 +18,64 @@
 Node.js Environment Setup
 =========================
 
+Contributing to the UI in Airflow
+-----------------------------------------
```
Comment (kaxil) [inline contributing-docs/14_node_environment_setup.rst]:
```suggestion
Contributing to the UI in Airflow
--------------------------------
```

File: contributing-docs/14_node_environment_setup.rst [hunk 2]
```diff
@@ -18,6 +18,64 @@
...
+1. Bug fixes to cherry pick for 2.10.x and 2.11
+2. The minimum necessary to unblock other Airflow 3.0 feature work
+3. Fixes to react views which we haven't migrated over yet, but can still be ported over to the new UI
+
+Custom endpoints for the UI will also be moved away from ``airflow/www/views.py`` and to ``airflow/api_ui``.
+Contributions to the legacy views file will follow the same rules.
+Committers will exercise their judgement on what endpoints should exist in the public ``airflow/api_connexion`` versus the private ``airflow/api_ui``
+
+Airflow UI
+-----------------------------------------
```
Comment (kaxil) [inline contributing-docs/14_node_environment_setup.rst]:
```suggestion
Airflow UI
----------
```

File: contributing-docs/14_node_environment_setup.rst [hunk 3]
```diff
@@ -18,6 +18,64 @@
...
+
+``airflow/ui`` is our React frontend powered by Vite with dependencies managed by pnpm.
```
Comment (kaxil) [inline contributing-docs/14_node_environment_setup.rst]: What's Vite? Is it https://vitejs.dev/ ?

File: contributing-docs/14_node_environment_setup.rst [hunk 4]
```diff
@@ -18,6 +18,64 @@
...
+Make sure you are using recent versions of pnpm\>=9 and node/>=20. ``breeze start-airflow`` will build the UI automatically.
```
Comment (kaxil) [inline contributing-docs/14_node_environment_setup.rst]:
```suggestion
Make sure you are using recent versions of ``pnpm>=9`` and ``node>=20``. ``breeze start-airflow`` will build the UI automatically.
```

File: contributing-docs/14_node_environment_setup.rst [hunk 5]
```diff
@@ -18,6 +18,64 @@
...
+Adding the ``--dev-mode`` flag will automatically run the vite dev server for hot reloading the UI during local development.
+
+pnpm commands
+-----------------------------------------
```
Comment (kaxil) [inline contributing-docs/14_node_environment_setup.rst]:
```suggestion
pnpm commands
-----------------
```

**Review by kaxil — APPROVED** [review]
Minor details. Worth dropping a note on the dev mailing list too to make sure committers have seen it.

File: contributing-docs/14_node_environment_setup.rst (hunk 1 above)
Comment (jscheffl) [reply contributing-docs/14_node_environment_setup.rst]: Too short, must be: :-D
```suggestion
Contributing to the UI in Airflow
---------------------------------
```

File: contributing-docs/14_node_environment_setup.rst (hunk 3 above)
Comment (jscheffl) [reply contributing-docs/14_node_environment_setup.rst]: Also asked myself about this and first thought I'm too stupid. THanks Kaxil for asking :-D (I was too shy)

File: contributing-docs/14_node_environment_setup.rst (hunk 5 above)
Comment (jscheffl) [reply contributing-docs/14_node_environment_setup.rst]: Right length is:
```suggestion
pnpm commands
-------------
```

**Comment by potiuk** [comment]
Nice. Other than the comments from @jscheffl and @kaxil -> no more from me.

File: contributing-docs/14_node_environment_setup.rst (hunk 3 above)
Comment (bbovenzi) [reply contributing-docs/14_node_environment_setup.rst]: Added a link and description.

**Review by kaxil — APPROVED** [review]

File: contributing-docs/14_node_environment_setup.rst [hunk 6]
```diff
@@ -18,6 +18,64 @@
 Node.js Environment Setup
 =========================
 
+Contributing to the UI in Airflow
```
Comment (jscheffl) [inline contributing-docs/14_node_environment_setup.rst:21]: I wanted to try and also found the docs in `airflow/ui/CONTRIBUTING.md` - does it make sense to cross-link here and remove the redundancy?

**Review by jscheffl — APPROVED** [review]
Made local setup and tested... works. Got an empty UI now :-D

**Review by eladkal — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

**Review by pierrejeambrun — APPROVED** [review]

**Pull request merged**